Assets/HoloToolkit/Utilities/Scripts/Extensions/TransformExtensions.cs

```diff
@@ -27,13 +29,64 @@ private static void GetFullPath(StringBuilder stringBuilder, Transform transform
             if (transform.parent == null)
             {
                 stringBuilder.Append(prefix);
-            }
-            else
+            } else
             {
                 GetFullPath(stringBuilder, transform.parent, delimiter, prefix);
                 stringBuilder.Append(delimiter);
             }
             stringBuilder.Append(transform.name);
         }
+
+        /// <summary>
+        /// Iterates the root object and all its children, using a queue rather than recursively.
+        /// </summary>
+        /// <param name="root">Start point of the traversion set</param>
+        public static IEnumerable<Transform> IterateHierarchy(this Transform root)
+        {
+            if (root == null) { throw new ArgumentNullException("root"); }
+            return root.IterateHierarchyCore(new List<Transform>(0));
+        }
+
+        /// <summary>
+        /// Iterates the root object and all its children except for the branches in ignore, using a queue rather than recursively.
+        /// </summary>
+        /// <param name="root">Start point of the traversion set</param>
+        /// <param name="ignore">Transforms and all its children to be ignored</param>
+        public static IEnumerable<Transform> IterateHierarchy(this Transform root, ICollection<Transform> ignore)
+        {
+            if (root == null) { throw new ArgumentNullException("root"); }
+            if (ignore == null)
+            {
+                throw new ArgumentNullException("ignore", "Ignore collection can't be null, use IterateHierarchy(root) instead.");
+            }
+            return root.IterateHierarchyCore(ignore);
+        }
+
+        /// <summary>
+        /// Iterates the root object and all its children except for the branches in ignore, using a queue rather than recursively.
+        /// </summary>
+        /// <param name="root">Start point of the traversion set</param>
+        /// <param name="ignore">Transforms and all its children to be ignored</param>
+        private static IEnumerable<Transform> IterateHierarchyCore(this Transform root, ICollection<Transform> ignore)
+        {
+            var parentsQueue = new Queue<Transform>();
+            parentsQueue.Enqueue(root);
+
+            while (parentsQueue.Count > 0)
+            {
+                var parent = parentsQueue.Dequeue();
+
+                if (ignore.Contains(parent)) { continue; }
+
+                foreach (Transform child in parent)
```

Comment:
Oh I see, parent isn't a `Transform` Component. 

Partly why I avoid `var` when types are not evident. No worries then.